odiglet/pkg/kube/runtime_details/shared.go

```diff
@@ -64,33 +64,42 @@ func runtimeInspection(pods []corev1.Pod) ([]common.LanguageByContainer, error)
 				log.Logger.V(0).Info("no processes found in pod container", "pod", pod.Name, "container", container.Name, "namespace", pod.Namespace)
 				continue
 			}
+			if len(processes) > 1 {
+				// Currently we don't support multiple processes in the same container, where each one can have a different language
+				// We only take the first process into account, when we'll support multiple processes we'll need to change this.
+				log.Logger.V(0).Info("multiple processes found in pod container, only taking the first one into account", "pod", pod.Name, "container", container.Name, "namespace", pod.Namespace)
+			}
+			process := processes[0]
 
-			detectionResults := inspectors.DetectLanguage(processes)
-			if len(detectionResults) == 0 {
-				log.Logger.V(0).Info("no supported language detected for container in pod", "processes", processes, "pod", pod.Name, "container", container.Name, "namespace", pod.Namespace)
+			lang := inspectors.DetectLanguage(process)
+			if lang == common.UnknownProgrammingLanguage {
```

Comment:
perhaps instead of returning unknown, we can return nil? I suspect this unknown can have some side effects